src/Linux/Host/Broker/Impl/Services/LinuxSystemInfoService.cs

```diff
@@ -111,8 +118,103 @@ public class LinuxSystemInfoService : ISystemInfoService {
             }
         }
 
-        public (string VideoCardName, long VideoRAM, string VideoProcessor) GetVideoControllerInformation() {
-            return ("", 0, "");
+
+        private static object _videoCardInfoLock = new object();
+        private static List<VideoCardInfo> _videoCardInfo;
+        public IEnumerable<VideoCardInfo> GetVideoControllerInformation() {
+            lock (_videoCardInfoLock) {
+                if(_videoCardInfo != null) {
+                    return _videoCardInfo;
+                }
+
+                InitializeVideoCardInfo();
+                return _videoCardInfo;
+            }
+        }
+
+        private void InitializeVideoCardInfo() {
+            _videoCardInfo = new List<VideoCardInfo>();
+            const string vramPart = "VRAM:";
+            var lshwDisplayData = ExecuteAndGetOutput("/usr/bin/lshw", "-class display").Select(s => s.Trim()).ToArray();
+            var dmesgData = ExecuteAndGetOutput("dmesg", null).Where(s => s.ContainsIgnoreCase(vramPart)).ToArray();
+
+            string productPart = "product: ";
+            string vendorPart = "vendor: ";
+            string busPart = "bus info: ";
+
+            // Match anything that looks like 1024M or 1024MB
+            Regex pattern = new Regex("[0-9]+[GM][B]?", RegexOptions.IgnoreCase);
+
+            int count = 1;
+            for (int i=0;  i < lshwDisplayData.Length; ++i) {
```

Comment:
This whole thing feels like it could be a single regex with named matches to extract values that we need.